modules/api/src/main/java/org/apache/ignite/tx/TransactionOptions.java

```diff
@@ -39,10 +39,17 @@ public long timeoutMillis() {
     /**
      * Sets transaction timeout, in milliseconds.
      *
-     * @param timeoutMillis Transaction timeout, in milliseconds.
+     * @param timeoutMillis Transaction timeout, in milliseconds. Cannot be negative; 0 means 'use default timeout'.
+     *     For RO transactions, the default timeout is data availability time configured via ignite.gc.lowWatermark.dataAvailabilityTime
+     *     configuration setting.
+     *     For RW transactions, timeouts are not supported yet. TODO: IGNITE-15936
      * @return {@code this} for chaining.
      */
     public TransactionOptions timeoutMillis(long timeoutMillis) {
```

Comment:
Should we introduce lower and upper limits for tx timeout? It seems it's not valid to have a RO transaction with timeout > dataAvailabilityTime. Curious what will be a reasonable upper limit for RW transaction. There's similar discussion in https://docs.google.com/document/d/1jhDMWGm6XJ7VQDZ0U1BxXuAForssig_Xw8Hr3GwzD-4/edit?tab=t.0#heading=h.jse5gwdl8j40

Reply:
>  not valid to have a RO transaction with `timeout > dataAvailabilityTime`

Low Watermark is supposed to be locked for the duration of an RO transaction, so we should be able to exceed the data availability time, right?

P.S. The linked document is not public.

Reply:
Let's decide this out of this PR (which is already pretty big) as this matter doesn't seem to be a blocker. I filed an issue https://issues.apache.org/jira/browse/IGNITE-24240